doc/reference/fortran.rst

```diff
@@ -118,3 +118,180 @@ The basic error handler is an allocatable derived type, available from ``mctc_en
+      
+      !> Dispersion virial
+      real(wp), intent(out), contiguous, optional :: sigma(:, :)
+
+      class(damping_param), allocatable :: param
+      class(dispersion_model), allocatable :: disp
+
+      ! Get D4 damping parameters for given method
+      call get_rational_damping(method, param, s9=1.0_wp)
+      if (.not.allocated(param)) then
+         call fatal_error(error, "No parameters for '"//method//"' available")
+         return
+      end if
+
+      ! Initialize D4/D4S model
+      call new_dispersion_model(error, d4, mol)
+      if (allocated(error)) return
+      
+      call get_dispersion(mol, disp, param, realspace_cutoff(), energy, &
+         & gradient, sigma)
+
+   end subroutine calc_dftd4
+
+
+Complete Example
+----------------
+
+A minimal program using the snippets from above could look like this:
+
+.. code-block:: fortran
+
+   program demo
+      use, intrinsic :: iso_fortran_env, only : error_unit
+      use mctc_env, only : wp, error_type, fatal_error
+      use mctc_io, only : new, structure_type
+      use dftd4, only : damping_param, get_rational_damping, get_dispersion
+      use dftd4_cutoff, only : realspace_cutoff
+      use dftd4_model, only : dispersion_model, new_dispersion_model
+      use dftd4_utils, only : lowercase
+      implicit none
+   
+      !> Error handling
+      type(error_type), allocatable :: error
+   
+      type(structure_type) :: mol
+      real(wp) :: energy
+      real(wp), allocatable :: gradient(:, :), sigma(:, :)
+   
+      call ch4(mol)      
+      allocate(gradient(3, mol%nat), sigma(3, 3))
+   
+      call calc_dftd4(error, mol, "pbe", energy, gradient=gradient, sigma=sigma)
+   
+      if (allocated(error)) then
+         write(error_unit, '("[Error]:", 1x, a)') error%message
+         error stop
+      end if
+   
+      write (*,'(a,f18.12)') "D4 dispersion energy (Hartree): ", energy
+   
+   
+   contains
+   
+   
+   subroutine ch4(mol)
+
+      !> Molecular structure data
+      type(structure_type), intent(out) :: mol
+   
+      real(wp), allocatable :: xyz(:, :)
+      integer, allocatable :: num(:)
+   
+      num = [6, 1, 1, 1, 1]
+      xyz = reshape([ &
+        &  0.00000000000000_wp, -0.00000000000000_wp,  0.00000000000000_wp, &
+        & -1.19220800552211_wp,  1.19220800552211_wp,  1.19220800552211_wp, &
+        &  1.19220800552211_wp, -1.19220800552211_wp,  1.19220800552211_wp, &
+        & -1.19220800552211_wp, -1.19220800552211_wp, -1.19220800552211_wp, &
+        &  1.19220800552211_wp,  1.19220800552211_wp, -1.19220800552211_wp],&
+        & [3, size(num)])
+   
+      call new(mol, num, xyz, charge=0.0_wp, uhf=0)
+   
+   end subroutine ch4
+   
+   
+   subroutine calc_dftd4(error, mol, method, energy, model, gradient, sigma)
+   
+      !> Error handling
+      type(error_type), allocatable, intent(out) :: error
+   
+      !> Molecular structure data
+      type(structure_type), intent(in) :: mol
```

Comment:
This part also will have a lot of whitespace and requires a lot of scrolling in the webpage, let's make it more compact.